### DISCUSSION TITLE
How to fetch messages with setting the SEEN flag?

### DISCUSSION THREAD

**Discussion by jsuto** [mjs/imapclient#536] [Q&A] [answered]
I have a script that imports a user's emails to an archive daily by fetching today's data. However the user is confused because after the script has run all of his emails are marked as read (=seen). So I wonder of there's a way to instruct fetch not to set the SEEN flag?

**Comment by mjs** [comment, marked as answer]
There's two ways to do this, open the folder as read only when calling `select_folder` or retrieve messages using `BODY.PEEK[]`.

e.g.

```python
client.fetch(msg_ids, ['BODY.PEEK[]'])
```

`BODY.PEEK[]` is the same `RFC822` but won't affect the read status of messages.

https://www.rfc-editor.org/rfc/rfc3501#section-6.4.5
